### DISCUSSION TITLE
Add name or id to input element of Checkbox

### DISCUSSION THREAD

**Discussion by stabildev** [radix-ui/primitives#2439] [Help] [answered]
In Chrome dev tools under the issues tab I get the warning "A form field element should have an id or name attribute" for all Checkbox components. How can I set a name or id for the hidden input element (not the button)?

Edit: I have the same issue with the Select component.

**Comment by benoitgrelard** [comment, marked as answer]
If you pass a `name` to it, it will be added to that native `input` when inside a `form`.

**Comment by KPI01** [comment]
I am having the issue with the `select` element and the `id` attribute; the `name` attribute works just fine but I would wanted to set and `id` for testing purposes and I'm not able to set the attribute

**Reply by mbroadhead-bytes** [reply]
https://github.com/shadcn-ui/ui/issues/2459#issuecomment-2254364271
